lib/SILOptimizer/Mandatory/LowerHopToActor.cpp

```diff
@@ -106,23 +106,72 @@ bool LowerHopToActor::processHop(HopToExecutorInst *hop) {
   return true;
 }
 
+static bool isDefaultActorType(CanType actorType, ModuleDecl *M,
+                               ResilienceExpansion expansion) {
+  if (auto cls = actorType.getClassOrBoundGenericClass())
+    return cls->isDefaultActor(M, expansion);
+  return false;
+}
+
+static AccessorDecl *getUnownedExecutorGetter(ASTContext &ctx,
+                                              ProtocolDecl *actorProtocol) {
+  for (auto member: actorProtocol->getAllMembers()) {
+    if (auto var = dyn_cast<VarDecl>(member)) {
+      if (var->getName() == ctx.Id_unownedExecutor)
+        return var->getAccessor(AccessorKind::Get);
+    }
+  }
+  return nullptr;
+}
+
 SILValue LowerHopToActor::emitGetExecutor(SILLocation loc, SILValue actor) {
   // This is okay because actor types have to be classes and so never
   // have multiple abstraction patterns.
   CanType actorType = actor->getType().getASTType();
 
   auto &ctx = F->getASTContext();
-  auto builtinName = ctx.getIdentifier(
-    getBuiltinName(BuiltinValueKind::BuildSerialExecutorRef));
-  auto builtinDecl = cast<FuncDecl>(getBuiltinValueDecl(ctx, builtinName));
   auto resultType = SILType::getPrimitiveObjectType(ctx.TheExecutorType);
-  auto subs = SubstitutionMap::get(builtinDecl->getGenericSignature(),
-                                   {actorType}, {});
 
-  // Use builSerialExecutorRef to cast the actor to Builtin.Executor.
-  // TODO: use the Executor protocol
-  auto unmarkedExecutor =
-    B.createBuiltin(loc, builtinName, resultType, subs, {actor});
+  // If the actor type is a default actor, go ahead and devirtualize here.
```

Comment:
That's pretty cool and in practice the huge majority of actors right? Except obj-c compatible ones and if someone overrides e.g. the executor since that'll trigger `hasExplicitCustomActorMethods` right?

Reply:
We expect so, yes, with the major exception of the main actor.